### PR TITLE
added tests to verify invalid function parameters are handled correctly

### PR DESCRIPTION
**Pull request by dhiltonp**
Tests for #186.

### CODE REVIEW

**Comment by stevenh** [comment]
@dhiltonp could you just fix the missing new line at the end of the file please.

**Comment by dhiltonp** [comment]
Done

File: function_test.go [hunk 1]
```diff
@@ -278,3 +280,62 @@ func TestFunction_toString(t *testing.T) {
...
+func TestBadArgs(t *testing.T) {
+	vm := New()
+	vm.Set("http", map[string]interface{}{
+		"Get": http.Get,
+		"Post": http.Post,
+		"PostForm": http.PostForm,
+		"NewRequest": http.NewRequest,
+	})
+
+	_, e := vm.Run(`console.log(http.Post("https://httpbin.org/post"));`)
```
Comment (stevenh) [inline function_test.go]: Could you make this self hosting so it doesn't require an external connection please.

httptest package makes this easy, see: https://golang.org/pkg/net/http/httptest/

**Comment by stevenh** [comment]
@dhiltonp any news on this?

**Comment by dhiltonp** [comment]
I haven't looked at this at all, sorry.

Is #186 still an issue problem?

**Comment by stevenh** [comment]
> I haven't looked at this at all, sorry.
> 
> Is #186 still an issue problem?

I don't believe so see https://github.com/robertkrimen/otto/pull/419

**Comment by stevenh** [comment]
Super seeded by https://github.com/robertkrimen/otto/pull/419

**Pull request closed without merging**